src/deploy/osp_deployer/director.py

```diff
@@ -276,6 +276,17 @@ def node_discovery(self):
             raise AssertionError(
                 "Number of nodes in instackenv.json does not add up"
                 " to the number of nodes defined in .properties file")
+        if setts.numa_enable is True or setts.hpg_enable is True:
+            grep_data = self.run_tty("grep -n pm_addr ~/instackenv.json")[0].rstrip()
+            instack_nodes = list(grep_data.split(","))
+            compute_nodes = self.settings.compute_nodes
+            for compute_node in compute_nodes:
+                for instack_node in instack_nodes:
+                    if instack_node.find(compute_node.idrac_ip) > 0:
+                        extracted_line = int(instack_node.split(':')[0])
+                        cmd = "sed -i "+""+str(extracted_line+2)+"s/pxe_drac/pxe_ipmitool/"+" ~/instackenv.json"
```

Comment:
The above line is changing the Ironic driver from the iDRAC driver to the IPMI driver.  While it is true that a side affect of this is that in-band introspection will be run on the node, this change will cause other major problems, like:
- We won't do RAID configuration of the node
- We won't do BIOS configuration of the node
This is because the IPMI driver doesn't support either RAID or BIOS configuration.
Because of this, we should always use the Ironic iDRAC (pxe_drac) driver and never use the IPMI driver.